packages/languages/src/node/language-server-contribution.ts

```diff
@@ -26,12 +28,24 @@ export {
 export const LanguageServerContribution = Symbol('LanguageServerContribution');
 export interface LanguageServerContribution extends LanguageContribution {
     start(clientConnection: IConnection): void;
+    /**
+     * Called on shut down. Only synchronous operations are permitted.
+     */
+    stop?(): void;
```

Comment:
this API is misleading

`start` method is called each time when a new connection is established to start a language server, it can happen multiple times for the same client, e.g. the connection got lost and `LanguageClient` decided to restart the server.

I would expect that `stop` means to stop a language server and it should be called each time when the connection got lost. For this, though there is already `clientConnectiont.onClose` event, which btw can be used to remove a process from the process manager.

I think the following API would be better: 
```ts
start(clientConnection: IConnection): Disposable;
```

than a returned disposable can be disposed when the connection got lost as well as `onStop`.

Reply:
Thanks for the note, that's a leftover.